### PR TITLE
Review

### PR DESCRIPTION
**Pull request by psurkov**

### CODE REVIEW

File: src/main/java/com/plugin/frege/runConfiguration/FregeRunConfigurationFactory.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.plugin.frege.runConfiguration;
+
+import com.intellij.execution.configurations.ConfigurationFactory;
+import com.intellij.execution.configurations.ConfigurationType;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.openapi.project.Project;
+
+public class FregeRunConfigurationFactory extends ConfigurationFactory {
```
[the remark this answers is not in the document]
Comment (23jura23) [reply src/main/java/com/plugin/frege/runConfiguration/FregeRunConfigurationFactory.java:8]: Oh, this is the class that need to be removed (and the whole `runConfiguration` too), it is not used at all. I thought I've done this long ago...

File: src/main/kotlin/com/plugin/frege/psi/impl/FregePsiClassImpl.kt [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        return PsiClass.EMPTY_ARRAY // TODO
+    }
+
+    override fun findFieldByName(@NonNls name: String, checkBases: Boolean): PsiField? {
+        return null // TODO
+    }
+
+    override fun findMethodBySignature(patternMethod: PsiMethod, checkBases: Boolean): PsiMethod? {
+        val methods = findMethodsBySignature(patternMethod, checkBases)
+        return if (methods.isNotEmpty()) methods[0] else null
```
Comment (satamas) [inline src/main/kotlin/com/plugin/frege/psi/impl/FregePsiClassImpl.kt:110]: firstOrNull()

File: src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+     */
+    @JvmStatic
+    fun <T : PsiElement> getByTypePredicateCheckingText(
+        clazz: KClass<out PsiElement>, element: T, incompleteCode: Boolean
+    ): (elem: PsiElement?) -> Boolean {
+        val instancePredicate = { elem: PsiElement? -> clazz.isInstance(elem) }
+        return if (incompleteCode) {
+            instancePredicate
+        } else {
+            elem: PsiElement? -> instancePredicate(elem) && elem != null && elem.text == element.text
```
Comment (satamas) [inline src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt:73]: From my point of view a more complex branch should be the first one

File: src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    }
+
+    /**
+     * @return [FregeWhereSection] if the first found binging contains it or `null` it not.
+     */
+    @JvmStatic
+    fun findWhereInExpression(element: PsiElement): FregeWhereSection? {
+        val binding = getParentBinding(element)
+        val elements = findElementsWithinElementSequence(binding) { elem: PsiElement -> elem == element }
+        return if (elements.any()) {
```
Comment (satamas) [inline src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt:84]: Why `any`?

Comment (kkarnauk) [reply src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt:84]: It's a check if our element is in the scope of the first binding-parent.

Comment (satamas) [reply src/main/kotlin/com/plugin/frege/psi/impl/FregePsiUtilImpl.kt:84]: Ok, in a sequence it's an adequate method, I thought that it's a collection

**Pull request closed without merging**